src/Symfony/Component/HttpKernel/HttpCache/Store.php

```diff
@@ -317,14 +317,30 @@ private function getMetadata($key)
     /**
      * Purges data for the given URL.
      *
+     * This method purges both the HTTP and the HTTPS version of the cache entry.
+     *
      * @param string $url A URL
      *
-     * @return bool true if the URL exists and has been purged, false otherwise
+     * @return bool true if the URL exists with either HTTP or HTTPS scheme and has been purged, false otherwise
      */
     public function purge($url)
     {
-        $key = $this->getCacheKey(Request::create($url));
+        $http = preg_replace('#^https#', 'http', $url);
+        $https = preg_replace('#^http#', 'https', $url);
 
+        return $this->doPurge($http) || $this->doPurge($https);
```

Comment:
Based on the title of this PR, I would have expected this to be an `&&` instead of an `||`